src/checkers/flymake-collection-statix.el

```diff
@@ -0,0 +1,134 @@
+;;; flymake-collection-statix.el --- Statix diagnostic function -*- lexical-binding: t -*-
+
+;; Copyright (c) 2022 Valeriy Litkovskyy
+
+;; Permission is hereby granted, free of charge, to any person obtaining a copy
+;; of this software and associated documentation files (the "Software"), to deal
+;; in the Software without restriction, including without limitation the rights
+;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+;; copies of the Software, and to permit persons to whom the Software is
+;; furnished to do so, subject to the following conditions:
+
+;; The above copyright notice and this permission notice shall be included in all
+;; copies or substantial portions of the Software.
+
+;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+;; SOFTWARE.
+
+;;; Commentary:
+
+;; This file provides a flymake backend that lets the user analyze nix code
+;; using statix.
+
+;;;; Tips
+
+;; + You can customize `flymake-collection-statix-args' to your liking.
+
+;; + Some warnings have "{fix}" suffix.  They can be fixed by calling
+;;   `flymake-collection-statix-fix'.
+
+;;;; Credits
+
+;; This diagnostic would not have been possible without the statix[1] program.
+
+;;  [1] https://github.com/nerdypepper/statix
+
+;;; Code:
+
+;;;; Requirements
+
+(require 'flymake)
+(require 'flymake-collection)
+
+(eval-when-compile
+  (require 'flymake-collection-define))
+
+;;;; Customization
+
+(defcustom flymake-collection-statix-args nil
+  "Additional statix check arguments."
+  :type '(repeat :tag "Arguments" (string :tag "Argument"))
+  :group 'flymake-collection)
+
+;;;; Commands
+
+(defun flymake-collection-statix-fix ()
+  "Fix region at point using statix suggestion."
+  (interactive)
+  (when-let ((data (cl-loop for diagnostic in (flymake-diagnostics (point))
+                            for data = (flymake-diagnostic-data diagnostic)
+                            when data return data)))
+    (save-restriction
+      (widen)
+      (let-alist data
+        (delete-region .beg .end)
+        (insert .fix)))))
+
+;;;; Functions
+
+;;;;; Public
+
+;;;###autoload (autoload 'flymake-collection-statix "flymake-collection-statix")
+(flymake-collection-define-enumerate flymake-collection-statix
```

Comment:
Some new tests for this would be great.

Reply:
Hmmm... that's a pickle. Ordinarily you'd install it through the package manager but statix isn't in the apt archive for ubuntu 20.04. When that doesn't work I normally just curl a relase tarball from github that has it already compiled (example: [lf](https://github.com/gokcehan/lf/releases/tag/r27)). That won't work here either since statix doesn't include a binary release (note: Might be worth creating an issue about that, for compiled programs and common architectures this is very convenient). Last resort would be setting up the toolchain for rust in the container itself and then building it after the fact or preferably setup a [multi-stage](https://docs.docker.com/build/building/multi-stage/) build where you compile statix from a base image that already has rust installed and then just copy the binary into the real final container. The latter would be what I would recommend, even if it kind of side-steps the installers directory configuration.